app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -259,6 +259,13 @@ class Product extends \Magento\ImportExport\Model\Import\Entity\AbstractEntity
      */
     protected $_mediaGalleryAttributeId = null;
 
+    /**
+     * Existing product ids in product import
+     *
+     * @var array
+     */
+    private $existingProductIds = [];
```

Comment:
`$this->existingProductIds` is not being reset.
You should probably reset it in `_saveProducts()` when new `getNextBunch()` is taken